controllers/nutanixmachine_controller.go

```diff
@@ -475,23 +480,23 @@ func (r *NutanixMachineReconciler) reconcileNode(rctx *nctx.MachineContext) erro
 	patchHelper, err := patch.NewHelper(node, remoteClient)
 	if err != nil {
 		klog.Errorf("%s Failed to create patchHelper for the workload cluster node %s. %v", rctx.LogPrefix, nodeName, err)
-		return err
+		return nil, err
 	}
 
 	node.Spec.ProviderID = rctx.NutanixMachine.Spec.ProviderID
 	err = patchHelper.Patch(rctx.Context, node)
 	if err != nil {
 		klog.Errorf("%s Failed to patch the remote workload cluster node %s's spec.providerID. %v", rctx.LogPrefix, nodeName, err)
-		return err
+		return nil, err
 	}
 	klog.Infof("%s Patched the workload node %s spec.providerID: %s", rctx.LogPrefix, nodeName, node.Spec.ProviderID)
 
-	return nil
+	return nil, nil
```

Comment:
imo, this return a `reconcile.Result{}, nil` ?

Reply:
We should not return a reconcile.Result since we don't want to trigger a new reconciliation loop. Hence it is nil.